heron/api/src/scala/org/apache/heron/streamlet/scala/Builder.scala

```diff
@@ -49,7 +50,14 @@ trait Builder {
     * Creates a new Streamlet using the underlying generator
     *
     * @param generator The generator that generates the tuples of the streamlet
-    * @return  a Streamlet representation of the source object
+    * @return a Streamlet representation of the source object
     */
   def newSource[R](generator: Source[R]): Streamlet[R]
+
+  /**
+   * Creates a new Streamlet using the provided spout
+   * @param spout The spout that emits the tuples of the streamlet
+   * @return a Streamlet representation of the spout object
+   */
+  def newSource[R](spout: IRichSpout): Streamlet[R]
```

Comment:
@nwangtw Thanks for working on this PR.
**Integration Test** is also useful for end2end coverage. 

Currently, the following streamlet ITs exist: 
- **Python:** `word_count_streamlet` 
- **Scala:** `ScalaStreamletWithFilterAndTransform` and `ScalaStreamletWithMapAndFlatMapAndFilterAndClone`
- **Java Streamlet API** does not have any IT yet.

Reply:
Agreed. And there could be some bugs in the API since so far it seems people are testing it but I haven't heard anyone using it in prod yet.

Let me prepare another PR to at least add the two Scala tests to java. https://github.com/apache/incubator-heron/issues/3101

And we definitely need better IT coverage for Streamlet API.

Reply:
@nwangtw Thanks for following this.
I developed Scala Integration Tests. If you want, i can also address `Java ITs` in the light of #3101. Also, please let me know to collaborate on other Streamlet Improvements ;)